services/brig/test/integration/API/User/Auth.hs

```diff
@@ -76,12 +98,56 @@ tests conf m z b = testGroup "auth"
 --------------------------------------------------------------------------------
 -- ZAuth test environment for generating arbitrary tokens.
 
-randomAccessToken :: ZAuth ZAuth.AccessToken
-randomAccessToken = randomUserToken >>= ZAuth.newAccessToken
 
-randomUserToken :: ZAuth ZAuth.UserToken
+randomAccessToken :: forall u a . ZAuth.TokenPair u a => ZAuth (ZAuth.Token a)
+randomAccessToken = randomUserToken @u >>= ZAuth.newAccessToken
+
+randomUserToken :: ZAuth.UserTokenLike u => ZAuth (ZAuth.Token u)
 randomUserToken = (Id <$> liftIO UUID.nextRandom) >>= ZAuth.newUserToken
 
+-------------------------------------------------------------------------------
+-- Nginz authentication tests (end-to-end sanity checks)
+--
+
+testNginz :: Brig -> Nginz -> Http ()
+testNginz b n = do
+    -- Note: If you get 403 test failures: check that the private/public keys in brig and nginz match.
+    u <- randomUser b
+    let Just email = userEmail u
+    -- Login with email
+    rs <- login b (defEmailLogin email) PersistentCookie
+        <!! const 200 === statusCode
+    let c = decodeCookie rs
+        t = decodeToken rs
+    -- ensure nginz allows refresh at /access
+    _rs <- post (n . path "/access" . cookie c . header "Authorization" ("Bearer " <> (toByteString' t))) <!! do
+        const 200 === statusCode
```

Comment:
I think this is the line that the comment in 114 refers to.  You could move the comment here (not tested):

```suggestion
        assertEqual "/access (requires that private key on nginz and public key on brig match)"
            (const 200) statusCode
```

Reply:
The Bilge.assert and the HUnit.assert don't work well together, so the above doesn't compile, but I can do this:

```haskell
_rs <- get (n . path "/clients" . header "Authorization" ("Bearer " <> (toByteString' t)))
liftIO $ assertEqual "Ensure nginz is started. Ensure nginz and brig share the same private/public zauth keys. Ensure ACL file is correct." 200 (statusCode _rs)
```

Reply:
Oops, I thought there was an `assertEqual` in hunit and one in bilge.  You could you https://hoogle.zinfra.io/file/wire-server/.stack-work/install/x86_64-linux/custom-wire-1.1-G6NVtOVxqIB2/8.4.3/doc/bilge-0.22.0/Bilge-Assert.html#v:assertTrue, though?